pkg/util/phase.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha1"
+
+	kmapi "kmodules.xyz/client-go/api/v1"
+)
+
+func PhaseFromCondition(conditions []kmapi.Condition) api.DatabasePhase {
+	// Generally, the conditions should maintain the following chronological order
+	// For normal restore process:
+	//   ProvisioningStarted --> ReplicaReady --> AcceptingConnection --> DataRestoreStarted --> DataRestored --> Ready --> Provisioned
+	// For restoring the volumes (PerconaXtraDB):
+	//	 ProvisioningStarted --> DataRestoreStarted --> DataRestored --> ReplicaReady --> AcceptingConnection --> Ready --> Provisioned
+
+	// These are transitional conditions. They can update any time. So, their order may vary:
+	// 1. ReplicaReady
+	// 2. AcceptingConnection
+	// 3. DataRestoreStarted
+	// 4. DataRestored
+	// 5. Ready
+	// 6. Paused
+
+	var phase api.DatabasePhase
+
+	// ==================================  Handling "ProvisioningStarted" condition  ========================
+	// If the condition is present and its "true", then the phase should be "Provisioning".
+	if kmapi.IsConditionTrue(conditions, api.DatabaseProvisioningStarted) {
+		phase = api.DatabasePhaseProvisioning
+	}
+
+	// ================================= Handling "ReplicaReady" condition ==========================
+	// If the condition is present and its "false", then the phase should be "Critical".
+	if kmapi.HasCondition(conditions, api.DatabaseReplicaReady) && !kmapi.IsConditionTrue(conditions, api.DatabaseReplicaReady) {
```

Comment:
IsConditionFalse ?

Reply:
No. Otherwise, when the database is being provisioned and the StatefulSets are being created, the database phase would be `Critical` which is not desired. That time database phase should be `Provisioning` or `NotReady`.

Reply:
ok. you have added new method which do the same.